receiver/sshcheckreceiver/factory_windows.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+// +build windows
+
+package sshcheckreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/configssh"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/metadata"
+)
+
+const (
+	typeStr   = "sshcheck"
+	stability = component.StabilityLevelInDevelopment
+)
+
+// NewFactory creates a new receiver factory
+func NewFactory() component.ReceiverFactory {
+	return component.NewReceiverFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithMetricsReceiver(createMetricsReceiver, stability))
+}
+
+func createDefaultConfig() config.Receiver {
+	return &Config{
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			ReceiverSettings:   config.NewReceiverSettings(config.NewComponentID(typeStr)),
+			CollectionInterval: 10 * time.Second,
+		},
+		SSHClientSettings: configssh.SSHClientSettings{
+			Timeout: 10 * time.Second,
+		},
+		Metrics: metadata.DefaultMetricsSettings(),
+	}
+}
+
+// on Windows we just want to inform that it's presently unsupported
+func createMetricsReceiver(_ context.Context, params component.ReceiverCreateSettings, rConf config.Receiver, consumer consumer.Metrics) (component.MetricsReceiver, error) {
+	return nil, errWindowsUnsupported
+}
+
+// this function is redefined here strictly for the build to succeed when certain test files referencing it aren't excluded
+// even though those tests are gated per os
+func newScraper(conf *Config, settings component.ReceiverCreateSettings) *sshcheckScraper {
+	return &sshcheckScraper{
+		Config:   conf,
+		settings: settings.TelemetrySettings,
+		mb:       metadata.NewMetricsBuilder(conf.Metrics, settings.BuildInfo),
+	}
+}
+
+// this function is redefined here strictly for the build to succeed when certain test files referencing it aren't excluded
+// even though those tests are gated per os
+func timeout(deadline time.Time, timeout time.Duration) time.Duration {
```

Comment:
Since the goal for handling this was to return informative errors on Windows, using the factory to detect this at receiver load seemed like as good of an option. It accomplished the direct goal of simplifying code to address this and then some further simplification.